amethyst_assets/src/asset.rs

```diff
@@ -19,19 +21,19 @@ use StoreId;
 /// for audio data.
 ///
 /// This data is then generated by the `Format` trait.
-pub trait Asset
-where
-    Self: Clone + Sized,
-{
+pub trait Asset: Clone + Sized {
     /// The `Context` type that can produce this asset
-    type Context: Context<Asset=Self, Data=Self::Data, Error=Self::Error>;
+    type Context: Context<Asset=Self, Data=Self::Data, Error=Self::Error, Result=Self::Result>;
 
     /// The `Data` type the asset can be created from.
     type Data;
 
     /// The error that may be returned from `Self::Context::from_data`.
     type Error: Error;
 
+    /// The result type of `Self::Context::create_asset`
+    type Result: IntoFuture<Item=Self, Error=Self::Error>;
```

Comment:
I created a typedef for the `AssetError` then since it will be `<A as Blah>::` like 4 times then..